salt/cloud/clouds/azurearm.py

```diff
@@ -533,10 +533,10 @@ def _get_node_info(node):
                 image_ref['sku'],
                 image_ref['version'],
             ])
-        except TypeError:
+        except KeyError:
```

Comment:
Are we certain we don't want to catch both instead of changing from one to the other?

Reply:
I don't see how it would throw TypeError ever... but I can change it if you think it's best.